### PR TITLE
Sync Salesforce with KYB data 

### PR DESCRIPTION
**Pull request by MatanYadaev**
On completion:
image

### CODE REVIEW

File: services/workflows-service/src/events/workflow-completed-webhook-caller.ts [hunk 1]
```diff
@@ -120,4 +126,36 @@ export class WorkflowCompletedWebhookCaller {
...
+    }
+
+    const statusMap = {
+      [StateTag.APPROVED]: 'Approved',
+      [StateTag.REJECTED]: 'Rejected',
+    } as const;
+
+    let status: (typeof statusMap)[keyof typeof statusMap] | undefined;
+
+    (workflowRuntimeData.tags ?? []).forEach((tag: string) => {
```
Comment (Omri-Levy) [inline services/workflows-service/src/events/workflow-completed-webhook-caller.ts]:
```suggestion
    workflowRuntimeData?.tags?.forEach((tag: string) => {
```

File: services/workflows-service/src/events/workflow-completed-webhook-caller.ts [hunk 2]
```diff
@@ -120,4 +126,36 @@ export class WorkflowCompletedWebhookCaller {
...
   }
+
+  private async updateSalesforceRecord(workflowRuntimeData: WorkflowRuntimeData) {
+    const { salesforceRecordId, salesforceObjectName } = workflowRuntimeData;
+
+    if (!salesforceRecordId || !salesforceObjectName) {
+      return;
+    }
+
+    const statusMap = {
```
Comment (Omri-Levy) [inline services/workflows-service/src/events/workflow-completed-webhook-caller.ts:137]: Thoughts about using an array here instead? Could use `some` or `includes` and such. Most of our enums also export a type and an array.

Comment (Omri-Levy) [reply services/workflows-service/src/events/workflow-completed-webhook-caller.ts:137]: Picking `some` or `includes` here over `forEach` on `tags` itself might also fit more.

File: services/workflows-service/src/workflow/workflow.controller.external.ts [hunk 3]
```diff
@@ -318,6 +318,8 @@ export class WorkflowControllerExternal {
       currentProjectId,
     );
 
+    const hasSalesforceRecord = 'salesforceObjectName' in body && 'salesforceRecordId' in body;
```
Comment (Omri-Levy) [inline services/workflows-service/src/workflow/workflow.controller.external.ts]: What if their value is `undefined` or an empty string?

File: services/workflows-service/src/workflow/workflow.service.ts [hunk 4]
```diff
@@ -1884,4 +1907,23 @@ export class WorkflowService {
...
   }
+
+  async updateSalesforceRecord({
+    workflowRuntimeData,
+    data,
+  }: {
+    workflowRuntimeData: WorkflowRuntimeData;
+    data: {
+      KYB_Started_At__c?: Date;
+      KYB_Status__c?: 'Not Started' | 'In Progress' | 'Approved' | 'Rejected';
```
Comment (Omri-Levy) [inline services/workflows-service/src/workflow/workflow.service.ts:1924]: Should we extract to an enum?

File: services/workflows-service/src/workflow/workflow.controller.external.ts (hunk 3 above)
Comment (MatanYadaev) [reply services/workflows-service/src/workflow/workflow.controller.external.ts]: Empty string is ok, but `undefined` is not. It's a client input, how can it be set to `undefined`?

File: services/workflows-service/src/workflow/workflow.service.ts (hunk 4 above)
Comment (MatanYadaev) [reply services/workflows-service/src/workflow/workflow.service.ts:1924]: I don't think so...

File: services/workflows-service/src/workflow/workflow.controller.external.ts (hunk 3 above)
Comment (Omri-Levy) [reply services/workflows-service/src/workflow/workflow.controller.external.ts]: Let me re-phrase. Is validation not needed there beyond the existence of property?

Comment (MatanYadaev) [reply services/workflows-service/src/workflow/workflow.controller.external.ts]: We didn't enable `exactOptionalPropertyTypes` so I changed the condition a bit. Thanks.

**Review by tomer-shvadron — APPROVED** [review]

**Pull request merged**